File: php/src/Domain/Books/Novel.php
<?php

namespace Katalogs\BookInvoicing\Domain\Books;

use Katalogs\BookInvoicing\Domain\Country\Language;

class Novel implements Book
{

    public function __construct(
        public readonly string $name,
        public readonly float $price,
        public readonly Author $author,
        public readonly Language $language,
        Genre ...$genres
    ) { }

    public function getAuthor(): Author
    {
        return $this->author;
    }
}

File: java/src/main/java/org/peaksys/gameoftechs/domain/book/Novel.java
package org.peaksys.gameoftechs.domain.book;


import com.fasterxml.jackson.annotation.JsonProperty;
import org.peaksys.gameoftechs.domain.country.Language;

import java.util.List;

public record Novel(
        String name,
        double price,
        Author author,
        Language language,
        @JsonProperty("genre")
        List<Genre> genres
) implements Book {
}
